File: src/server/AnkaCMS.Service/Models/PartModel.cs
using AnkaCMS.Core;
using AnkaCMS.Core.ValueObjects;
using System;
using System.Collections.Generic;

namespace AnkaCMS.Service.Models
{
    public class PartModel : IServiceModel
    {

        public Guid Id { get; set; }
        public int DisplayOrder { get; set; }
        public bool IsApproved { get; set; }
        public int Version { get; set; }
        public DateTime CreationTime { get; set; }
        public IdCodeName Creator { get; set; }
        public DateTime LastModificationTime { get; set; }
        public IdCodeName LastModifier { get; set; }
        public string Code { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Keywords { get; set; }
        public Guid PartId { get; set; }
        public IdCodeName Language { get; set; }
        public List<IdCodeNameSelected> Contents { get; set; }

    }
}

File: src/client/angular-web-admin/src/app/models/part-model.ts
import { IdCodeName } from '../value-objects/id-code-name';
import { IdCodeNameSelected } from '../value-objects/id-code-name-selected';

export class PartModel {
    id: string;
    displayOrder: number;
    isApproved: boolean;
    version: number;
    creationTime: Date;
    creator: IdCodeName;
    lastModificationTime: Date;
    lastModifier: IdCodeName;
    code: string;
    name: string;
    description: string;
    keywords: string;
    language: IdCodeName;
    partId: string;
    maxItemCount: number;
    contents: IdCodeNameSelected[];
    constructor() {
        this.creator = new IdCodeName();
        this.lastModifier = new IdCodeName();
        this.language = new IdCodeName();
        this.contents = [];
    }
}
